src/modules/Event/page.tsx

```diff
@@ -19,32 +19,66 @@ import {
 import {
   Card,
   CardContent,
-  CardHeader,
   CardLabel,
   CardSeparator,
-  CardTitle,
 } from "@/shared/components/card";
-import { PulseIcon } from "@cartridge/ui";
-
-const DisplayFormat = ["hex", "dec", "string"] as const;
+import { PulseIcon, cn } from "@cartridge/ui";
+import {
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+} from "@/shared/components/tabs";
+import { CopyableInteger } from "@/shared/components/copyable-integer";
+import FeltDisplayAsToggle from "@/shared/components/FeltDisplayAsToggle";
 
 export function Event() {
   const {
-    data: { eventId, txHash, receipt, block, event, eventData, eventKeys },
+    data: { eventId, txHash, receipt, block, event, decodedEvent },
     isLoading,
     error,
   } = useEvent();
+
+  if (error) {
+    console.error("Error fetching event", error);
+  }
+
   const { isMobile } = useScreen();
-  console.log(error);
 
   const [formats, setFormats] = useState<{
-    keys: DisplayFormatTypes;
-    data: DisplayFormatTypes;
-  }>({ keys: "hex", data: "hex" });
+    rawKeys: DisplayFormatTypes;
+    rawData: DisplayFormatTypes;
+    decodedKeys: DisplayFormatTypes;
+    decodedData: DisplayFormatTypes;
+  }>({
+    rawKeys: "hex",
+    rawData: "hex",
+    decodedKeys: "hex",
+    decodedData: "hex",
+  });
+
+  const [activeTab, setActiveTab] = useState<"raw" | "decoded">("raw");
+
+  const hasRawKeys = event?.keys.length > 0;
+  const hasRawData = event?.data.length > 0;
+  const hasDecodedKeys = Boolean(decodedEvent?.keys?.length);
+  const hasDecodedData = Boolean(decodedEvent?.data?.length);
+  const hasDecoded = Boolean(
+    decodedEvent && (hasDecodedKeys || hasDecodedData),
+  );
+
+  useEffect(() => {
```

Comment:
I don't know enough react to evaluate if this is correct or not. Maybe need some review here @nrmnqdds.